pkgs/pre-commit-check/default.nix

```
@@ -89,7 +89,19 @@ pre-commit-hooks.lib.${system}.run {
               if [ "$PRE_COMMIT" = "1" ] \
               && [ -n "$PRE_COMMIT_FROM_REF" ] \
               && [ -n "$PRE_COMMIT_TO_REF" ]; then
-                ${config.hooks.commitizen.package}/bin/cz check --rev-range "$PRE_COMMIT_FROM_REF".."$PRE_COMMIT_TO_REF"
+
+                # If the hook runs in a merge queue, we only allow Revert prefixes,
+                # to avoid merging with (fixup!, squash!, Merge, ..) commits.
+                # The $IS_MERGE_QUEUE variable is set by the merge-queue action in CI.
+                ALLOWED_PREFIXES_FLAG=""
+                ALLOWED_PREFIXES=("Revert")
+                if [ "$IS_MERGE_QUEUE" = "1" ]; then
+                  ALLOWED_PREFIXES_FLAG="--allowed-prefixes \"''${ALLOWED_PREFIXES[@]}\""
```

Are the single quotes stray?
```suggestion
                  ALLOWED_PREFIXES_FLAG="--allowed-prefixes \"${ALLOWED_PREFIXES[@]}\""
```

---

No, it's a nix string so we have to escape the ${..} or nix will try to substitute the inner expression.

---

https://nix.dev/manual/nix/2.25/language/string-literals#:~:text=A%20%22dollar%2Dcurly%22%20($%7B)%20can%20be%20written%20as%20follows:

---

Ahhh 😜